imap_processing/codice/codice_l1a.py

```diff
@@ -335,6 +342,42 @@ def unpack_science_data(self, science_values: str) -> None:
                 )
             )
 
+    def set_data_product_config(
+        self, apid: int, packet: xr.Dataset, data_version: str
+    ) -> None:
+        """
+        Set the various settings for defining the data products.
+
+        Parameters
+        ----------
+        apid : int
+            The APID of interest.
+        packet : xarray.Dataset
+            A packet for the APID of interest.
+        data_version : str
+            Version of the data product being created.
+        """
+        self.packet_dataset = packet
+
+        # Gather and set various configurations of the data product
+        config = constants.DATA_PRODUCT_CONFIGURATIONS.get(apid)  # type: ignore[call-overload]
+        self.coords_to_include = config["coords"]
```

Comment:
Would it make sense to set a `self.config = config`, and then lookup these values later since they are all named identically. `self.config["dataset_name"]` rather than `self.dataset_name` doesn't seem too bad.